packages/storage-plus/src/indexed_map.rs

```diff
@@ -730,6 +793,225 @@ mod test {
         assert_eq!(datas[1], marias[1].1);
     }
 
+    #[test]
+    #[cfg(feature = "iterator")]
+    fn range_de_simple_string_key() {
+        let mut store = MockStorage::new();
+        let map = build_map();
```

Comment:
Here it would be better to test it against a map with a pk that does not deserialize to `Vec<u8>`. Either `String`, `&str`, or even better, one Uint key (`Uint8` or `Uint32`), would work.

You would have to either adapt the non-deserializing tests, or create a different map here (i.e. a `build_map_de()` helper). I think the first option is slightly more work but it's better, as we'll be able to clearly show the difference between `range` and `range_de` result types.